src/check-update.ts

```diff
@@ -0,0 +1,250 @@
+import { BUILDER_TAG, HOOKS_TAG, RUNTIME_TAG } from "./mod.ts";
+
+const DENO_SLACK_SDK = "deno_slack_sdk";
+const DENO_SLACK_API = "deno_slack_api";
+const DENO_SLACK_HOOKS = HOOKS_TAG.substring(0, HOOKS_TAG.indexOf("@"));
+const DENO_SLACK_BUILDER = BUILDER_TAG.substring(0, BUILDER_TAG.indexOf("@"));
+const DENO_SLACK_RUNTIME = RUNTIME_TAG.substring(0, RUNTIME_TAG.indexOf("@"));
+const IMPORT_MAP_SDKS = [DENO_SLACK_SDK, DENO_SLACK_API];
+const SLACK_JSON_SDKS = [
+  DENO_SLACK_HOOKS, // should be the only one needed once the get-scripts hook is supported
+  DENO_SLACK_BUILDER, // but just in case, we can look for builder and runtime as well!
+  DENO_SLACK_RUNTIME,
+];
+
+interface UpdateResponse {
+  update: boolean;
+  breaking: boolean;
+  message: string;
+  error: {
+    message: string;
+  };
+}
+
+interface VersionMap {
+  [key: string]: {
+    current: string;
+    latest: string;
+    update: boolean;
+    breaking: boolean;
+    error: string;
+  };
+}
+
+export const checkForSDKUpdates = async () => {
+  const versionMap: VersionMap = await createVersionMap();
+  const updateResp = createUpdateResp(versionMap);
+  return updateResp;
+};
+
+/**
+ * createVersionMap creates an object that contains each dependency,
+ * featuring information about the current and latest versions, as well
+ * as if breaking changes are present and if any errors occurred during
+ * version retrieval.
+ */
+async function createVersionMap() {
+  const versionMap: VersionMap = await readProjectDependencies();
+
+  // Check each dependency for updates, classify update as breaking or not,
+  // craft message with information retrieved, and note any error that occurred.
+  for (const [sdk, value] of Object.entries(versionMap)) {
+    if (value) {
+      const current = versionMap[sdk].current;
+      let latest = "", error = null;
+
+      try {
+        latest = await fetchLatestModuleVersion(sdk);
+      } catch (err) {
+        error = err;
+      }
+
+      const update = current !== latest;
+      const breaking = hasBreakingChange(current, latest);
+
+      versionMap[sdk] = {
+        ...versionMap[sdk],
+        latest,
+        update,
+        breaking,
+        error,
+      };
+    }
+  }
+
+  return versionMap;
+}
+
+/** readProjectDependencies reads from possible dependency files
+ * (import_map.json, slack.json) and maps them to the versionMap
+ * containing each dependency's update information
+ */
+async function readProjectDependencies(): Promise<VersionMap> {
+  const cwd = Deno.cwd();
+  const versionMap: VersionMap = {};
+
+  // Find SDK component versions in import map, if available
+  const map = await getJson(`${cwd}/import_map.json`);
```

Comment:
This will probably work most of the time since it's the default name for the import map we use. If we want to support it w/o a few edge cases, we could look for a `deno.json` or `deno.jsonc` file, and look for an `importMap` property set in there, which is ultimately what deno would end up using as the import map (they could name it anything they wanted).

Reply:
Update on this: @filmaj and I put together a solution that accomplished the suggested improvement, but discovered that the current method in place that reads the file doesn't handle the presence of comments that might be found in a `deno.jsonc`. 

Rather than quickly pull in a new dependency to see it ship, we agreed to circle back when we have more time to devote to ensuring it works as expected in all cases.